tests/test_tools.py

```diff
@@ -729,34 +730,49 @@ def test_group_member_ids(input, expected):
     actual = group_member_ids(input, "bg")
     assert actual == expected
 
-def test_fragment_retrieval():
-    p1 = {
-        "_id": "scopatz",
-        "aka": [
-            "Scopatz",
-            "Scopatz, A",
-            "Scopatz, A.",
-            "Scopatz, A M",
-            "Anthony Michael Scopatz",
-        ],
-        "name": "Anthony Scopatz",
-    }
-    p2 = {
-        "_id": "abc",
-        "aka": [
-            "A. BC",
-            "BC, A",
-            "Anthony BC",
-        ],
-        "name": "Anthony Bill Chris",
-    }
-    assert fragment_retrieval([p1, p2], ["aka", "name", "_id"],
-                           "Anthony") == [p1, p2]
-    assert fragment_retrieval([p1,p2], ["aka", "name", "_id"],
-                           "scopatz, a", case_sensitive = True) == []
-    assert (
-            fragment_retrieval(
-                [p1,p2], ["aka", "name", "_id"], "scopatz, a"
-            )
-            == [p1]
-    )
\ No newline at end of file
+p1 = {
+    "_id": "scopatz",
+    "aka": [
+        "Scopatz",
+        "Scopatz, A",
+        "Scopatz, A.",
+        "Scopatz, A M",
+        "Anthony Michael Scopatz",
+    ],
+    "name": "Anthony Scopatz",
+}
+p2 = {
+    "_id": "abc",
+    "aka": [
+        "A. BC",
+        "BC, A",
+        "Anthony BC",
+    ],
+    "name": "Anthony Bill Chris",
+}
+
+@pytest.mark.parametrize(
+    "input1, input2, input3, input4, expected",
```

Comment:
maybe cleaner if you make a list or a tuple, ``inputs`` and then make the signature ``inputs[0], inputs[1]``